edx_rest_framework_extensions/tests/test_middleware.py

```diff
@@ -0,0 +1,93 @@
+"""
+Unit tests for safe_endpoints Middleware.
+"""
+import ddt
+from itertools import product
+
+from django.test import TestCase, RequestFactory
+from rest_framework.authentication import SessionAuthentication
+from rest_framework_jwt.authentication import BaseJSONWebTokenAuthentication
+from rest_framework.decorators import api_view
+from rest_framework.views import APIView
+
+from ..middleware import EnsureJWTAuthSettingsMiddleware
+from ..permissions import JwtHasScope
+
+
+class SomeIncludedPermissionClass(object):
+    pass
+
+
+class SomeJwtAuthenticationSubclass(BaseJSONWebTokenAuthentication):
+    pass
+
+
+def some_auth_decorator(include_jwt_auth=True, include_required_perm=True):
+    def _decorator(f):
+        f.permission_classes = (SomeIncludedPermissionClass,)
+        f.authentication_classes = (SessionAuthentication,)
+        if include_jwt_auth:
+            f.authentication_classes += (SomeJwtAuthenticationSubclass,)
+        if include_required_perm:
+            f.permission_classes += (JwtHasScope,)
+        return f
+    return _decorator
+
+
+@ddt.ddt
+class TestEnsureJWTAuthSettingsMiddleware(TestCase):
+    def setUp(self):
+        super(TestEnsureJWTAuthSettingsMiddleware, self).setUp()
+        self.request = RequestFactory().get('/')
+        self.middleware = EnsureJWTAuthSettingsMiddleware()
+
+    def _assert_included(self, item, iterator, should_be_included):
+        if should_be_included:
+            self.assertIn(item, iterator)
+        else:
+            self.assertNotIn(item, iterator)
+
+    @ddt.data(
+        *product(
+            (True, False),
+            (True, False),
+            (True, False),
+        )
+    )
+    @ddt.unpack
+    def test_api_views(self, use_function_view, include_jwt_auth, include_required_perm):
```

Comment:
You could also test that logging is happening when you expect it to.

Reply:
true.  updated.